### ISSUE TITLE
SqlCommand.Dispose doesn't free managed object

### ISSUE THREAD

**Issue by breyed** [dotnet/SqlClient#74]
`SqlCommand.Dispose` contains this code:

``` c#
// release managed objects
_cachedMetaData = null;
```

This doesn't free the cached data, which is supposed to be the purpose of the [Dispose(Boolean) overload](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#dispose-and-disposeboolean) when working with managed resources that don't implement `IDisposable`:
> **Managed objects that consume large amounts of memory or consume scarce resources.** Freeing these objects explicitly in the Dispose method releases them faster than if they were reclaimed non-deterministically by the garbage collector.

By just setting `_cachedMetaData` to null, the resources are only reclaimed by the garbage collector, by which time the disposed `SqlCommand` object would also likely not be referenced, meaning that disposing the `SqlCommand` and setting `_cachedMetaData` to null is wasted effort.

----

This issue was observed as part of a discussion about this question: [Is SqlCommand.Dispose() required if associated SqlConnection will be disposed?](https://stackoverflow.com/q/1808036/145173)

**Comment by divega** [comment]
As recently [announced in the .NET Blog](https://devblogs.microsoft.com/dotnet/introducing-the-new-microsoftdatasqlclient/), focus on new SqlClient features an improvements is moving to the new [Microsoft.Data.SqlClient package](https://www.nuget.org/packages/Microsoft.Data.SqlClient/). For this reason, we are moving this issue to the new repo at https://github.com/dotnet/SqlClient. We will still use https://github.com/dotnet/corefx to track issues on other providers like System.Data.Odbc and System.Data.OleDB, and general ADO.NET and .NET data access issues.

**Comment by mdaigle** [comment]
I agree with the sentiment, but unfortunately we're bound by the API set in System.Data.Common.DBCommand. System.Data.Common types enable the same API when accessing via ODBC and OLEDB drivers. Moving away from System.Data.Common is a massive change that would span multiple drivers and disrupt a huge number of people.
